scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -658,14 +658,24 @@ protected boolean createExecution(NodeSet nodeSet, Node node, InternalJob job, I
 
                     tlogger.debug(task.getId(), "deploying");
 
+                    // Dynamically adjust the start-task-timeout according to the number dependency tasks in a merge.
+                    // above 500 parent tasks, it is worth adjusting.
+                    if (taskDescriptor.getParents().size() > 500) {
```

Comment:
and 500 could be specified into a global variable

Reply:
I would, in my next PR...

Reply:
Or could it be interesting to consider it as a user variable?

Reply:
it would be pointless:
the timeout we calculate depends on the number of parents. The objective is not the find the best timeout value, but to assign a value that guarantees that the task won't fail if its parents results take too long to be fetched.
By letting the user define the SCHEDULER_STARTTASK_TIMEOUT, we only multiply this value every 500 parents. 
The 500 value was chosen after dozens of tests, but it doesn't really matter if it's 500 or 1000 since we don't want to tweak that but only to have a big enough timeout value